src/domain/synthetics/positions/utils.ts

```diff
@@ -245,3 +246,47 @@ export function formatLeverage(leverage?: BigNumber) {
 
   return `${formatAmount(leverage, 4, 2)}x`;
 }
+
+export function getEstimatedLiquidationTime(
+  position: PositionInfo,
+  minCollateralUsd: BigNumber | undefined,
+  period = "1h"
+): number {
+  const { marketInfo, isLong, sizeInUsd, isOpening, netValue } = position;
+
+  if (isOpening || !minCollateralUsd) return 0;
+
+  let liquidationCollateralUsd = applyFactor(sizeInUsd, marketInfo.minCollateralFactor);
+  if (liquidationCollateralUsd.lt(minCollateralUsd)) {
+    liquidationCollateralUsd = minCollateralUsd;
+  }
+  const borrowFeePerHour = getBorrowingFeeRateUsd(marketInfo, isLong, sizeInUsd, CHART_PERIODS[period]);
+  const fundingFeePerHour = getFundingFeeRateUsd(marketInfo, isLong, sizeInUsd, CHART_PERIODS[period]);
+  const priceImpactDeltaUsd = getPriceImpactForPosition(marketInfo, sizeInUsd.mul(-1), isLong, {
+    fallbackToZero: true,
+  });
+
+  const totalFeesPerHour = borrowFeePerHour.abs().add(fundingFeePerHour.lt(0) ? fundingFeePerHour.abs() : 0);
+
+  if (totalFeesPerHour.eq(0)) return 0;
+
+  const hours = netValue
+    .add(priceImpactDeltaUsd.lt(0) ? priceImpactDeltaUsd : 0)
+    .sub(liquidationCollateralUsd)
+    .div(borrowFeePerHour.abs().add(fundingFeePerHour.abs()));
+
+  return hours.toNumber();
+}
+
+export function formatEstimatedLiquidationTime(hours: number) {
```

Comment:
hours seems like required parameter, but in common, i think formatter function should handle optinal params to simplify jsx code, so you can just change param to  hours?: number )

Reply:
Fixed!